lib/galaxy/schema/schema.py

```diff
@@ -266,16 +294,126 @@ def schema_extra(schema: Dict[str, Any], model) -> None:
+
+
+class LimitedUserModel(Model):
+    """This is used when config options (expose_user_name and expose_user_email) are in place."""
+
+    id: EncodedDatabaseIdField = UserIdField
+    username: Optional[str]
+    email: Optional[str]
+
+
+class DiskUsageUserModel(Model):
+    total_disk_usage: float = TotalDiskUsageField
+    nice_total_disk_usage: str = NiceTotalDiskUsageField
+
+
+class CreatedUserModel(UserModel, DiskUsageUserModel):
+    preferred_object_store_id: Optional[str] = PreferredObjectStoreIdField
+
+
+class AnonUserModel(DiskUsageUserModel):
+    quota_percent: Any = QuotaPercentField
+
+
+class DetailedUserModel(BaseUserModel, AnonUserModel):
+    is_admin: bool = Field(default=Required, title="Is admin", description="User is admin")
+    purged: bool = Field(default=Required, title="Purged", description="User is purged")
+    preferences: Dict[Any, Any] = Field(default=Required, title="Preferences", description="Preferences of the user")
+    preferred_object_store_id: Optional[str] = PreferredObjectStoreIdField
+    quota: str = Field(default=Required, title="Quota", description="Quota applicable to the user")
+    quota_bytes: Any = Field(
+        default=Required, title="Quota in bytes", description="Quota applicable to the user in bytes."
+    )
+    tags_used: List[str] = Field(default=Required, title="Tags used", description="Tags used by the user")
+
+
+class UserCreationPayload(Model):
+    password: str = Field(default=Required, title="user_password", description="The password of the user.")
+    email: str = UserEmailField
+    username: str = UserNameField
+
+
+class RemoteUserCreationPayload(Model):
+    remote_user_email: str = UserEmailField
+
+
+class UserDeletionPayload(Model):
+    purge: bool = Field(default=Required, title="Purge user", description="Purge the user")
+
+
+class FavoriteObject(Model):
+    object_id: str = Field(
+        default=Required, title="Object ID", description="The id of an object the user wants to favorite."
+    )
+
+
+class FavoriteObjectsSummary(Model):
+    tools: List[str] = Field(
+        default=Required, title="Favorite tools", description="The name of the tools the user favored."
+    )
+
+
+class FavoriteObjectType(str, Enum):
+    tools = "tools"
+
+
+class DeletedCustomBuild(Model):
+    message: str = Field(
+        default=Required, title="Deletion message", description="Confirmation of the custom build deletion."
+    )
+
+
+class CustomBuildBaseModel(Model):
+    name: str = Field(default=Required, title="Name", description="The name of the custom build.")
+
+
+# TODO Add descriptions
+class CustomBuildCreationPayload(CustomBuildBaseModel):
+    len_type: str = Field(
+        default=Required,
+        alias="len|type",
+        title="Length type",
+        description="TODO",
+    )
+    len_value: str = Field(
+        default=Required,
+        alias="len|value",
+        title="Length value",
+        description="TODO",
+    )
+
+
+# TODO Add descriptions
```

Comment:
So I was looking up what those things are ... `len` is the primary id of the len file, `count` is the number of chromosomes/contigs, `fasta` is the primary id of the fasta file from a history, `linecount` is the primary if of a linecount dataset. We should never return unencoded primary ids, and we don't consume the response in the client. I'm happy to merge this as is (including the TODOs here) and then follow up and remove the response, or if you want to just return a 204 status code that would be fine too.

Reply:
The primary id fields are now of type `EncodedDatabaseIdField` and the operation returns a 204 response like you suggested.

I also tried to add descriptions to the fields of this model. Could you please check if they are fitting?
https://github.com/galaxyproject/galaxy/blob/4e56a620afffa09414c155e1ad41df1821005f03/lib/galaxy/schema/schema.py#L392

Reply:
Thanks, that looks good to me!